homework4/WaitAndNotifyExample.java

```diff
@@ -0,0 +1,100 @@
+class Notifyclassbase extends Thread {
+    public void run() {
+        synchronized (this) {
+            System.out.println("Start " + Thread.currentThread().getName());
+            try {
+                System.out.println(Thread.currentThread().getName() + " waiting");
+                this.wait();
+
+            } catch (InterruptedException e) {
+            }
+
+            System.out.println(Thread.currentThread().getName() + " notified");
+            this.notify();
+        }
+    }
+}
+
+class Notify_thread2 extends Thread {
+    Notifyclassbase notify_obj;
+    int a = 0;
+
+    Notify_thread2(Notifyclassbase notify_obj) {
+        this.notify_obj = notify_obj;
+    }
+
+    public void run() {
+        synchronized (this.notify_obj) {
+            System.out.println("Start " + Thread.currentThread().getName());
+            try {
+                System.out.println(Thread.currentThread().getName() + " waiting");
+                this.notify_obj.wait();
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+            System.out.println(Thread.currentThread().getName() + " notified");
+        }
+    }
+}
+
+class Notify_thread3 extends Thread {
+    Notifyclassbase notify_obj;
+
+    Notify_thread3(Notifyclassbase notify_obj) {
+        this.notify_obj = notify_obj;
+    }
+
+    public void run() {
+        synchronized (this.notify_obj) {
+            System.out.println("Start " + Thread.currentThread().getName());
+            try {
+                System.out.println(Thread.currentThread().getName() + " waiting");
+                this.notify_obj.wait();
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+            System.out.println(Thread.currentThread().getName() + " notified");
+            this.notify_obj.notify();
+        }
+    }
+    
+}
+
+class Notify_thread4 extends Thread {
+    Notifyclassbase notify_obj;
+
+    Notify_thread4(Notifyclassbase notify_obj) {
+        this.notify_obj = notify_obj;
+    }
+
+    public void run() {
+        synchronized (this.notify_obj) {
+            System.out.println("Start " + Thread.currentThread().getName());
+            this.notify_obj.notify();
+            System.out.println(Thread.currentThread().getName() + "...notified");
+        }
+    }
+}
+
+
+public class WaitAndNotifyExample {
```

Comment:
Either use camel-case or snake case, do not use both in your code
with java, camel-case is used by default. which is the convention followed.